Release 0.4.0 details (#163)
submariner-io/submariner-website: src/content/releases/_index.en.md

weight = 25
+++

## v0.4.0 Libreswan cable driver, Kubernetes multicluster service discovery

> This release is mainly focused on the libreswan Submariner cable driver implementation, as well 
> as standardizing the service discovery support in Lighthouse with the upstream
> kubernetes KEP [1].

* The **libreswan** IPsec cable driver is available for testing. It's been included
  on the testing matrix of the project.
* **Lighthouse** has been modified per [1] as follows:
   - A `ServiceExport` object needs to be created alongside any `Service` that is intended to be exported to participant clusters
   - The supercluster services can now be accessed with `<service-name>.<namespace>.svc.supercluster.local`

> * **Globanet** overlapping CIDR support improvement and bug fixes

Author: ```suggestion
* **Globanet** overlapping CIDR support improvements and bug fixes
```

weight = 25
+++

> ## v0.4.0 Libreswan cable driver, k8s multicluster service discovery

Reviewer: Change k8s to Kubernetes

> ## v0.4.0 Libreswan cable driver, k8s multicluster service discovery

Author: ```suggestion
## v0.4.0 Libreswan cable driver, Kubernetes multicluster service discovery
```

> > This release is focused on the libreswan cabledriver implementation, as well 
> > as standardizing the service discovery support in lighthouse with the upstream
> > kubernetes kep [1].

Reviewer 2: ```suggestion
> This release is mainly focused on the libreswan Submariner cable driver implementation, as well 
> as standardizing the service discovery support in Lighthouse with the upstream
> kubernetes KEP [1].
```

> * The **libreswan** IPsec cabledriver is available for testing. It's been included

Reviewer: s/libreswan/Libreswan

  on the testing matrix of the project.

> * **Lighthouse** behavior has been modified following [1] for the next details:
>    - A `ServiceExport` object needs to be created alongside any `Service` that we
>      want exported to the participant clusters

Reviewer 2: ```suggestion
* **Lighthouse** has been modified per [1] as follows:
   - A `ServiceExport` object needs to be created alongside any `Service` that is intended to be exported to participant clusters
```

   - The supercluster services can now be accessed with `<service-name>.<namespace>.svc.supercluster.local`

> * **Globanet** overlapping CIDR support improvement and bugfixes

Reviewer: @sridhargaddam anything in particular we should highlight? At the minimum we should use "improvements" and not just one "improvement"

> * **Globanet** overlapping CIDR support improvement and bugfixes

Reviewer 3: yes agree @nyechiel, there were couple of improvements mostly around stability.

Having said it, we still have some known issues that needs to be handled and until then its possible that CI would occasionally fail. 
i.e., 
1. we have to update the e2e connector pod logic to wait until its Ready/globalIp-annotated before starting the tests.
2. the redundancy tests in e2e are only looking at submariner-gateway pod status and not submariner-globalnet status when gateway is migrated. This is occasionally causing some failures.
3. sometimes we are seeing errors in kube-proxy which would affect the functionality of cluster-ip services on the gateway node. I discussed this with @mangelajo and we wanted to try with some latest kube-proxy code (as the one used in KIND is very old).
